### PR TITLE
Add redux firsttime

### PR DESCRIPTION
**Pull request by MattGomb**
Dear reviewer!

In this PR I added some redux starter components. I installed the necessary dependencies, and created the suggested layout of folders and files.

I created books and categories, and i followed the ducks pattern.

I also tried to configure an initial 'store' component.

I would really appreciate some feedback about the logic, if I laid down the correct foundation, or some parts don't make sense maybe. I am especially curious about the addBook action - does it link with the 'Book' react component properly, or I should do something different?
Also, overall, does the naming make sense..?

Thanks in advance!

### CODE REVIEW

File: src/redux/books/books.js [hunk 1]
```diff
@@ -0,0 +1,31 @@
+import Book from '../../components/Book';
+
+// actions
+
+const ADD = 'ADD';
```
Comment (topeogunleye) [inline src/redux/books/books.js]: - [x] I kindly suggest that the value of each action type should follow the [ducks pattern](https://github.com/erikras/ducks-modular-redux). Please make changes to your remove book action type also

## EXAMPLE
```
const ADD_BOOK = bookstore/books/ADD_BOOK

const REMOVE_BOOK = bookstore/books/REMOVE_BOOK
```

File: src/redux/categories/categories.js [hunk 2]
```diff
@@ -0,0 +1,20 @@
+// actions
+
+const CHECK = 'CHECK';
```
Comment (topeogunleye) [inline src/redux/categories/categories.js]: - [x] I kindly suggest that the value of each action type should follow the [ducks pattern](https://github.com/erikras/ducks-modular-redux).

## EXAMPLE
```
const CHECK_STATUS  = bookstore/categories/CHECK_STATUS

```

File: src/redux/books/books.js [hunk 3]
```diff
@@ -0,0 +1,31 @@
...
+const REMOVE = 'REMOVE';
+
+// reducer
+
+export default function booksReducer(state = [], action) {
+  switch (action.type) {
+    case ADD:
+      return [...state, action.payload];
```
Comment (topeogunleye) [inline src/redux/books/books.js:13]: Great job here. the logic is correct. For your question, In order to link to the book react component, you'll have to first wrap your app in index.js with a provideR store which you'll learn about in future lessons, then you can use Use Selector to get objects from the provider store.

**Review by topeogunleye — CHANGES_REQUESTED** [review]
Hi @MattGomb, 

Good job so far!
There are some issues that you still need to work on to go to the next project but you are almost there!

Try-Again

> ## (Highlights) Good Points: 💪 
> * Good commit messages. :heavy_check_mark:
> * No linter errors. :heavy_check_mark:
> * Followed Github flow :heavy_check_mark:
> * Professional README file. :heavy_check_mark:
> * PR has a good title and summary :heavy_check_mark:

## Required Changes ♻️
_Check the comments under the review._

## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._

You can also consider:
-  Nothing to mention.

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag @topeogunleye in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

File: src/redux/books/books.js (hunk 1 above)
Comment (MattGomb) [reply src/redux/books/books.js]: Thanks for the suggestion, done!

File: src/redux/categories/categories.js (hunk 2 above)
Comment (MattGomb) [reply src/redux/categories/categories.js]: Thanks for the suggestion, done!

File: src/redux/books/books.js (hunk 3 above)
Comment (MattGomb) [reply src/redux/books/books.js:13]: Thanks for taking the time and check!

**Comment by MattGomb** [comment]
Seems like the ESLint doesn't want to finish, but i ran linters in VSCode, i fixed a typo, and everything was fine there

**Comment by topeogunleye** [comment]
> Seems like the ESLint doesn't want to finish, but i ran linters in VSCode, i fixed a typo, and everything was fine there

I can confirm there are nThe issue might be from GitHub. Maybe when you push again, it'll fix

> Seems like the ESLint doesn't want to finish, but i ran linters in VSCode, i fixed a typo, and everything was fine there

@MattGomb , it works now 👍

**Review by topeogunleye — APPROVED** [review]
### Hi @MattGomb, 

# Status: Approved ✔️
Excellent job on making the changes requested by the previous code reviewer. Keep it up 👏

Your project is complete! There is nothing else to say other than... it's time to merge it 💪
Congratulations! 🎉
congratulations
>
## To highlight :champagne: 
- All linters are passing :heavy_check_mark: 
- Good commit history :heavy_check_mark: 
- Followed Github flow :heavy_check_mark:
- Professional README file. :heavy_check_mark:
- PR has a good description :heavy_check_mark:
 
## [Optional] suggestions:
* Nothing to mention.
 
 _Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._
 
Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag @topeogunleye in your question so I can receive the notification.



------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Pull request merged**